### ISSUE TITLE
UmbracoMemberAuthorize attribute returns 401 after upgrade to 10.6.1 and does not redirect to login

### ISSUE THREAD

**Issue by jattwood** [umbraco/Umbraco-CMS#14555]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.6.1

### Bug summary

Pre-upgrade, from 10.4.x [UmbracoMemberAuthorize] controller decorator would redirect to a login page specified in `IConfigureNamedOptions<CookieAuthenticationOptions>`. After upgrade to 10.6.1, the redirect to login does not occur and browser returns 401 response only.

Looks like PR #14036 did some work on default behaviors for API only and broke page and surface controller redirect functionality. #14399 did some work to fix this. I don't think that these have been merged into the 10 branch as my versions don't seem to include any of these two PRs as of 10.6.1. 

I need this to work on 10.6.1 due to the security patch offered in 10.6.1 and looks like PR #14036 is a breaking change.

### Steps to reproduce

Follow the steps outlined here to protect a controller for a given page:

As outlined here: https://docs.umbraco.com/umbraco-cms/reference/routing/umbraco-api-controllers/authorization#using-memberauthorizeattribute

After decorating a hijacked page controller with the attribute, hitting that page while not logged in will result in a 401 unauthorized.

Previously we could redirect on 401 failure with the following:

In startup services:
` services.ConfigureOptions<ConfigureCustomMemberCookieOptions>();`

Create a class:

```
using Microsoft.AspNetCore.Authentication.Cookies;
using Microsoft.AspNetCore.Identity;
using Microsoft.Extensions.Options;

namespace Ninepoint.Core.Configuration
{
    public class ConfigureCustomMemberCookieOptions : IConfigureNamedOptions<CookieAuthenticationOptions>
    {
        public void Configure(string name, CookieAuthenticationOptions options)
        {
            if (name == IdentityConstants.ApplicationScheme || name == IdentityConstants.ExternalScheme)
            {
                Configure(options);
            }
        }

        public void Configure(CookieAuthenticationOptions options)
        {
            options.AccessDeniedPath = "/account/login";
            options.LoginPath = "/account/login";
        }
    }
}
```

### Expected result / actual result

Result would be redirection to the login page, actual result is a blank page with a 401

**Comment by elit0451** [comment]
Apologies for that, it seems like we've gotten the release tags wrong. The fix for this issue (https://github.com/umbraco/Umbraco-CMS/pull/14399) will be available in the upcoming `10.7.0` Umbraco version.

If you are unable to upgrade to `10.6.1` currently, for the security patch, check the [Workaround section](https://umbraco.com/blog/security-advisory-july-13-2023-security-patches-for-umbraco-cms-are-available/) from the advisory.

Thanks again for reaching out.

**Comment by jattwood** [comment]
Thank you

**Comment by IanIsFluent** [comment]
This doesn't seem to work on Umbraco 11.4.1.

I have a `RenderController` which is protected with `[UmbracoMemberAuthorize]` but it isn't redirecting anywhere. Adding the above doesn't seem to help.

**Comment by jattwood** [comment]
Yes, unfortunately that fix is not going to make it until 11.4.2. This should be listed as a breaking change IMO.

**Comment by IanIsFluent** [comment]
> Yes, unfortunately that fix is not going to make it until 11.4.2. This should be listed as a breaking change IMO.

I was just looking at the tags on the fix - are you sure its not until 11.4. _3_ ?

**Comment by IanIsFluent** [comment]
> Just looked again and you are right! I swear it said 11.4.2 a few minutes ago... I am really looking forward to that release as we are dead in the water until then.

😢 

For me, I have a content node and I could set the 'Restrict Public Access...' on that and make it redirect. But I'll have to do that for every `RenderController`.

**Comment by jattwood** [comment]
The way my protected content nodes are structured I am not able to use that approach. For now, I am injecting IMemberManager as _memberManager into my controller and doing a test/redirect right away after entering Index():

```
            if (!_memberManager.IsLoggedIn())
            {
                return Redirect("/account/login/");
            }
```
But now I have to do this for every controller too which will have to be undone once `[UmbracoMemberAuthorize]` comes back online :(

**Comment by AaronSadlerUK** [comment]
This is a rather big breaking bug in my opinion which should have been patched with a minor patch by now!

I have had to use the workaround provided by @jattwood to fix my site, this meant updating 23 files that were originally protected by this filter!

Several weeks on and 10.7.0 still does not have a release date... Nor has there been a minor patch released to address this issue...

**Comment by elit0451** [comment]
Hello everyone, apologies for the wait! We are expecting new releases for v10 and v11 within this or next week.

**Comment by AaronSadlerUK** [comment]
@elit0451 Has this been included in 10.7.0 as I cannot see it on the changelog page?
